src/test/publishertests.ts

```diff
@@ -150,31 +167,356 @@ describe('PublisherTests', function() {
+            runTitle: "",
+            publishRunAttachments: true,
+            fileNumber: "0",
+            releaseUri: "abc",
+            releaseEnvironmentUri: "xyz"
+        };
+        var reader = new trr.XUnitResultReader();
+        var resultsFile = path.resolve(__dirname, './testresults/xunitresults.xml');
+        var testRun;
+        
+        reader.readResults(resultsFile, runContext).then(function (res) {
+            testRun = res.testRun;
+            results = res.testResults;
+            
+            //Verifying the test run details
+            assert.strictEqual("XUnit Test Run debug mac", testRun.name);
+            //assert.equal("", testRun.startDate);
+            //assert.equal("", testRun.completeDate);
+            assert.equal("debug", testRun.buildFlavor);
+            assert.equal("mac", testRun.buildPlatform);
+            assert.equal("abc", testRun.releaseUri);
+            assert.equal("xyz", testRun.releaseEnvironmentUri);
+            assert.equal(21, testRun.build.id);
+            assert.strictEqual(true, testRun.automated);			
+            
+            
+            //Verifying Test Results
+            assert.strictEqual(4, results.length);
+            assert.strictEqual("FailingTest", results[0].automatedTestName);
+            assert.strictEqual("PassingTest", results[1].automatedTestName);
+            assert.strictEqual("tset2", results[2].automatedTestName);
+            assert.strictEqual("test1", results[3].automatedTestName);
+            done();
+        },
+        function(err) {
+            assert(false, 'XUnit Reader Failed! Details : ' + err.message);
+            done();
+        }); 
+    })
+    
+    it('results.publish : XUnit reader sanity check with run title', function(done) {
+        
+        var testRun;
+        var runContext: trp.TestRunContext = {
+            requestedFor: "userx",
+            buildId: "21",
+            platform: "mac",
+            config: "debug",
+            runTitle: "My Title",
+            publishRunAttachments: true,
+            fileNumber: "0",
+            releaseUri: "abc",
+            releaseEnvironmentUri: "xyz"
+        };
+        var resultsFile = path.resolve(__dirname, './testresults/xunitresults.xml');
+        var reader = new trr.XUnitResultReader();
+        //var testRunWithResults = reader.readResults(resultsFile, runContext);
+        reader.readResults(resultsFile, runContext).then(function (res) {
+            testRun = res.testRun;
+            
+            //Verifying the test run details
+            assert.strictEqual("My Title", testRun.name);			
+            done();
+        },
+        function(err) {
+            assert(false, 'XUnit Reader Failed! Details : ' + err.message);
+            done();
+        });		
+    })
+    
+    it('results.publish : XUnit reader sanity check with run title and file number', function(done) {
+        
+        var testRun;
+        var runContext: trp.TestRunContext = {
```

Comment:
dont you feel refactoring into a createtestruncontext with few params is a better option than copy/pasting it ?